dataland-backend/src/main/kotlin/org/dataland/datalandbackend/repositories/StoredCompanyRepository.kt

```diff
@@ -49,6 +51,67 @@ interface StoredCompanyRepository : JpaRepository<StoredCompanyEntity, String> {
+     * (Prefix-Matches are ordered before Center-Matches,
+     * e.g. when searching for "a" Allianz will come before Deutsche Bank)
+     */
+    @Query(
+        nativeQuery = true,
+        value =
+        "WITH filtered_text_results as (" +
+            // Fuzzy-Search Company Name
+            "(SELECT company_id, company_name AS company_name," +
+            " CASE " +
+            " WHEN company_name = :#{#searchString} THEN 10" +
+            " WHEN company_name ILIKE :#{escape(#searchString)}% ESCAPE :#{escapeCharacter()} THEN 5" +
+            " ELSE 1" +
+            " END match_quality " +
+            " FROM stored_companies" +
+            " WHERE company_name ILIKE %:#{escape(#searchString)}% ESCAPE :#{escapeCharacter()}" +
+            " ORDER BY match_quality DESC, company_id LIMIT 100)" +
+
+            " UNION " +
+            // Fuzzy-Search Company Alternative Name
+            " (SELECT " +
+            " stored_company_entity_company_id AS company_id," +
+            " stored_companies.company_name AS company_name," +
+            " CASE " +
+            " WHEN company_alternative_names = :#{#searchString} THEN 9" +
+            " WHEN company_alternative_names ILIKE :#{escape(#searchString)}% ESCAPE :#{escapeCharacter()} THEN 4" +
+            " ELSE 1 " +
+            " END match_quality " +
+            " FROM stored_company_entity_company_alternative_names" +
+            " JOIN stored_companies ON stored_companies.company_id = " +
+            " stored_company_entity_company_alternative_names.stored_company_entity_company_id  " +
+            " WHERE company_alternative_names ILIKE %:#{escape(#searchString)}% ESCAPE :#{escapeCharacter()}" +
+            " ORDER BY match_quality DESC, company_id LIMIT 100)" +
+
+            " UNION" +
+            // Fuzzy-Search Company Identifier
+            "(SELECT company_identifiers.company_id, stored_companies.company_name AS company_name," +
+            " CASE " +
+            " WHEN identifier_value = :#{#searchString} THEN 10" +
+            " WHEN identifier_value ILIKE :#{escape(#searchString)}% ESCAPE :#{escapeCharacter()} THEN 3" +
+            " ELSE 0" +
+            " END match_quality" +
+            " FROM company_identifiers" +
+            " JOIN stored_companies ON stored_companies.company_id = company_identifiers.company_id " +
+            " WHERE identifier_value ILIKE %:#{escape(#searchString)}% ESCAPE :#{escapeCharacter()} " +
+            " ORDER BY match_quality DESC, company_id LIMIT 100)) " +
+            // Combine Results
+            "SELECT filtered_text_results.company_id AS companyId," +
+            " MIN(filtered_text_results.company_name) AS companyName" +
+            " FROM filtered_text_results " +
+            " GROUP BY filtered_text_results.company_id" +
+            " ORDER BY MAX(filtered_text_results.match_quality) DESC, companyId" +
+            " LIMIT 100 ",
```

Comment:
I would indent this in one because it is not part of the WITH anymore.

Reply:
I fear this is not our decision but the linter's to make.